[maypfv] iP

maypfv opened the pull request:
# Blip Bot 🤖 
> "Life is too complicated not to be orderly." - Martha Stewart ([Source](https://dansilvestre.com/productivity-quotes/))

Blip helps you to stay organised and never miss tasks you need to do! Blip is,
- Text based
- Easy to learn
- ~Fast~ _SUPER FAST_ to use

All you need to do is,
1. Download Blip's latest release from [here](https://github.com/maypfv/ip/releases/tag/v0.1)
2. Open your terminal and navigate to where JAR file has been downloaded using the command ` cd path\to\file`, where `path\to\file` is the path to your JAR file
3. Execute `java -jar blip.jar`
4. Let Blip manage your tasks for you!

Here are some of the features of Blip:
- [x] Manage tasks of types _ToDo_, _Deadline_ and _Events
- [x] Alert to users the different errors with detailed messages
- [x] Search for tasks in the list using task names
- [ ] Managing deadlines (Coming Soon)
- [ ] Reminders (Coming Soon)


## 
Are you an aspiring Java programmer? You can use Blip to practice your Java skills too! Here is the `main` method for Blip:
```ruby
public static void main(String[] args) {
        new Blip(FILE_PATH).run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

maypfv committed:
Delete: Duke file

maypfv committed:
Add: Level 0

maypfv committed:
Add: Level 1

maypfv committed:
Added: Level 2

maypfv committed:
Added: Level 3

maypfv committed:
Added: Level 4 (First draft without inheritance)

maypfv committed:
New: Task class.
New: Deadline, Event, Todo task classes inherited from Task class.
Added: Level 4 (With inheritance)

maypfv committed:
Added: Level 5 Exception Handling

maypfv committed:
Added: Level 6 Delete Tasks

maypfv committed:
Automated Text UI Testing

maypfv committed:
Added: Level 7 Saving to File and Loading File (Draft 1)

maypfv committed:
Edited: Save and Load function implementations (Level 7)

maypfv committed:
Add: LocalDateTime stored in Deadline and Event tasks

maypfv committed:
Added: moreOOP Classes (Draft 1)

maypfv committed:
Added: MoreOOP (Draft 2)

maypfv committed:
Fix: MoreOOP (Draft 3)

maypfv committed:
Fix: MoreOOP (Draft 4 Final)

maypfv committed:
Fix: Small changes

maypfv committed:
Add: Organised into packages

maypfv committed:
Merge branch 'add-gradle-support'

maypfv committed:
Set-up: Gradle build and run

maypfv committed:
Add JUnit: Test cases for ToDo and DateTimeConverter

maypfv committed:
Fix: Small fixes on ToDoTest file

maypfv committed:
Fix: Jar File Problems

maypfv committed:
Add: JavaDoc on all classes and methods

maypfv committed:
Add: Following the coding standard

maypfv committed:
Resolve: Merge conflict

maypfv committed:
Resolve: More merge conflicts

maypfv committed:
Add: Find Command

maypfv committed:
Fix: Merge Conflicts

maypfv committed:
Add: JavaDoc for FindCommand class

Chandan8186 commented on src/main/java/blip/parser/BlipParser.java:
```diff
@@ -0,0 +1,153 @@
+                    return new FindCommand(findDescription);
+                default:
+                    return new InvalidCommand(input);
+            }
```
Perhaps the switch/case/default statements could be at the same indentation according to the coding standard?

Chandan8186 commented on src/main/java/blip/parser/DateConverter.java:
```diff
@@ -0,0 +1,32 @@
+import java.time.LocalDateTime;
+import java.util.regex.Pattern;
+import java.util.regex.Matcher;
+import blip.exceptions.*;
```
Perhaps individual imports could be made to match the coding standard?

Chandan8186 commented on src/main/java/blip/tasks/Task.java:
```diff
@@ -0,0 +1,93 @@
+    /**
+     * Description of the task.
+     */
+        protected String description;
```
Perhaps indentation could be set to 4 to match the other attributes of Task class?

Chandan8186 commented on src/test/java/blip/parse/DateConverterTest.java:
```diff
@@ -0,0 +1,32 @@
+    }
+
+    @Test
+    public void convertDateTimeTest_throwsException_success() throws DateTimeFormatException{
```
Perhaps a whitespace could be added before the opening bracket to match the coding standard?

Chandan8186 reviewed (commented):
Overall, the coding standard seems to be met in most cases. Perhaps for the relevant Javadocs comments, an empty line could be added between the description and parameters.

maypfv committed:
Add: New tasks

maypfv committed:
Fix: PR peer review comments fixed

WeeeHung commented on src/main/java/blip/commands/MarkCommand.java:
```diff
@@ -0,0 +1,46 @@
+    /**
+     * Index of task to mark as done.
+     */
+    int index;
```
could possibly specify the access modifiers here

WeeeHung commented on src/main/java/blip/exceptions/WrongNumberException.java:
```diff
@@ -0,0 +1,11 @@
+ * The WrongTaskNumberException class is an exception class
+ * that is thrown when the index of delete/mark/unmark commands are not valid.
+ */
+public class WrongNumberException extends Exception{
```
good that there are so many exceptions handled!

WeeeHung commented on src/main/java/blip/parser/BlipParser.java:
```diff
@@ -0,0 +1,153 @@
+        try {
+            String[] parts = input.split(" ");
+
+            String command = parts[0].toLowerCase();
```
good that command handles upper case as well

WeeeHung commented on src/main/java/blip/ui/BlipUI.java:
```diff
@@ -0,0 +1,151 @@
+    /**
+     * Prints the invalid task number error message.
+     */
+    public void showInvalidTaskNumErr() {
```
very user-friendly outputs!

WeeeHung commented on src/test/java/blip/parse/DateConverterTest.java:
```diff
@@ -0,0 +1,32 @@
+import java.time.LocalDateTime;
+public class DateConverterTest {
+    @Test
+    public void convertDateTimeTest_validDateTime_success() throws DateTimeFormatException {
```
good naming that matches the convention

maypfv committed:
Add: GUI Work in progress

maypfv committed:
Add: GUI Work in progress 2

maypfv committed:
Add: GUI Work in progress 3

maypfv committed:
Add: GUI Work in progress 4

maypfv committed:
Add: GUI Final

maypfv committed:
Test: GUI after merge to branch

maypfv committed:
Add assertions to ensure correctness of code

This commit adds assertions to the code for testing functionality
of classes like Blip, BlipParser, DateConverter, BlipStorage and
Task.

This will help to ensure the correctness of those classes and its
associated methods.

maypfv committed:
Improve Code Quality

I have made the following improvements:

- Imported only relevant classes from packages instead of whole
package
- Removed dead code such as run() in Blip
- Avoided empty catch blocks
- Removed the recycling of variable input of the containsCommand
in InvalidCommand class
- Use the default branch in switch cases in BlipParser

maypfv committed:
Merge pull request #1 from maypfv/branch-A-Assertions

Add assertions to ensure correctness of code

maypfv committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/blip/parser/BlipParser.java

maypfv committed:
Merge pull request #2 from maypfv/branch-A-CodeQuality

Improve Code Quality

maypfv committed:
Added: C-Priority

Draft 1
Some error with loading file, does not read Deadline and Event task

maypfv committed:
Add: Priority Extension

maypfv committed:
Add: Priority Extension Final

maypfv committed:
Finalising for iP Submission (Draft 1)

maypfv committed:
Add: Different BlipUser Icon and Blip Icons

maypfv committed:
Making UI better

maypfv committed:
Final Changes

maypfv committed:
Test for Ui.png

maypfv committed:
For Ui.png submission

maypfv committed:
Add: UserGuide

maypfv committed:
Fix: JavaDocs

maypfv committed:
Minor fix

maypfv committed:
Linked Blip main to Launcher

maypfv committed:
Final changes

maypfv committed:
Some fixes in Images

maypfv committed:
Edit: Bye command exits the program

maypfv committed:
Removed blip.jar that was wrongly committed to the repo.
